File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        if (points.length == 0) return 0;
        
        Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
        
        int count = 1;
        int prevEnd = points[0][1];
        
        for (int i = 1; i < points.length; i++) {
            if (points[i][0] > prevEnd) {
                count++;
                prevEnd = points[i][1];
            } else {
                prevEnd = Math.min(prevEnd, points[i][1]);
            }
        }
        
        return count;
    }
}

File: LeetCode/0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        n = len(points)
        points.sort()
        count = 1
        prevEnd = points[0][1]
        for i in range(1, n):
            if points[i][0] > prevEnd:
                count += 1
                prevEnd = points[i][1]
            else:
                prevEnd = min(prevEnd, points[i][1])
        return count